src/components/authorization/authorized.jsx

```diff
@@ -0,0 +1,12 @@
+import { Navigate, Outlet, useLocation } from "react-router-dom";
+import { useAuth } from "../../context/auth-context";
+
+function Authorized() {
+  const { authState } = useAuth();
+  const location = useLocation();
+  return (
+    <>{authState.isLoggedIn ? <Outlet /> : <Navigate to="/signin" from={{state: location}} replace />}</>
+  );
+}
```

Comment:
appreciation - good that you have created a separate component for authorization